src/computation/logic/sorted-reverse-dict.ts

```diff
@@ -0,0 +1,40 @@
+export class SortedReverseDict {
```

Comment:
Surely there is something in Lodash that handles this? Haha.

Reply:
Some of the functionality is found in Lodash, but for example the popTop function has a special behaviour not found there. I believe it would be no easier implementing the additional functionality using Lodash instead.